include/seqan/index/find2_index_approx_its.h

```diff
@@ -0,0 +1,401 @@
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of NVIDIA Corporation nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)sven.boenigk@fu-berlin.de HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Sven Boenigk <sven.boenigk@fu-berlin.de>
+// ==========================================================================
+// Approximate String matching via search schemes on a substring index.
+// ==========================================================================
+
+//NOTE (svnbgnk) this file has to be include as <seqan/index/find2_index_approx_its.h>
+#ifndef SEQAN_INDEX_FIND2_INDEX_APPROX_ITS_H_
+#define SEQAN_INDEX_FIND2_INDEX_APPROX_ITS_H_
+
+namespace seqan {
+// Compare potential occurrences directly to genome if the range on the index is small enough.
+template <typename TDelegateD,
+          typename TText, typename TIndex, typename TIndexSpec,
+          typename TNeedle,
+          size_t nbrBlocks,
+          typename TDir>
+inline void directSearch(TDelegateD & delegateDirect,
+                  Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                  TNeedle const & needle,
+                  uint32_t const needleLeftPos,
+                  uint32_t const needleRightPos,
+                  uint8_t const errors,
+                  OptimalSearch<nbrBlocks> const & s,
+                  uint8_t const blockIndex,
+                  TDir const & /**/)
+{
+    auto const & genome = indexText(*iter.fwdIter.index);
+    uint32_t needleL = length(needle);
+    uint32_t blocks = s.pi.size();
+
+    std::vector<uint32_t> blockStarts(blocks - blockIndex);
+    std::vector<uint32_t> blockEnds(blocks - blockIndex);
+    for(uint32_t j = blockIndex; j < s.pi.size(); ++j){
+        uint32_t blockStart = (s.pi[j] - 1 == 0) ? 0 : s.chronBL[s.pi[j] - 2];
```

Comment:
variable not necessary, just assign it directly to `blockStarts[j - blockIndex]`